File: linked-list-reversal/swapping-node-in-linked-list.js
import LinkedList from "./_utils/linked_list.js";
import LinkedListNode from "./_utils/linked_list_node.js";

export function swapNodes(head, k) {
	let curr = head;

	for (let i = 0; i < k - 1; i++) {
		curr = curr.next;
	}

	let left = curr;
	let right = head;

	while (curr.next) {
		curr = curr.next;
		right = right.next;
	}

	[left.data, right.data] = [right.data, left.data];
	return head;
}

File: linked-list-reversal/swapping-node-in-linked-list.py
from linked_list import LinkedList
from linked_list_node import LinkedListNode


def swap_nodes(head, k):
    curr = head

    for i in range(k - 1):
        curr = curr.next

    left = curr
    right = head

    while curr.next:
        curr = curr.next
        right = right.next

    left.data, right.data = right.data, left.data

    return head
